### PR TITLE
[CodeLlama]: simplify infilling with `<FILL_ME>`

### PR DESCRIPTION
**Pull request by pcuenca**

### CODE REVIEW

File: codellama.md [hunk 1]
```diff
@@ -151,52 +151,54 @@ This is a specialized task particular to code models. The model is trained to ge
...
+To use this feature successfully, you need to pay close attention to the format used to train the model for this task, as it uses special separators to identify the different parts of the prompt. Fortunately, transformers' `CodeLlamaTokenizer` makes this very easy, as demonstrated below:
 
 ```python
 from transformers import AutoTokenizer, AutoModelForCausalLM
 import transformers
 import torch
 
 model_id = "codellama/CodeLlama-7b-hf"
-tokenizer = AutoTokenizer.from_pretrained(model_id)
+tokenizer = CodeLlamaTokenizer.from_pretrained(model_id)
```
Comment (osanseviero) [inline codellama.md:162]: Isn't this tokenizer used by default with `AutoTokenizer`?

**Review by osanseviero — COMMENTED** [review]
Cool stuff!

File: codellama.md (hunk 1 above)
Comment (pcuenca) [reply codellama.md:162]: Actually, no, it was `LlamaFastTokenizer` in my case, iirc (and it doesn't have the special token or logic). Perhaps @ArthurZucker can clarify if it'd be possible for the class to be changed here? https://huggingface.co/codellama/CodeLlama-13b-Instruct-hf/blob/main/tokenizer_config.json#L25 (sorry if it's been discussed before).

Wouldn't that also make the changes available automatically in TGI?

Comment (pcuenca) [inline codellama.md:162]:
```suggestion
tokenizer = AutoTokenizer.from_pretrained(model_id)
```

Comment (pcuenca) [reply codellama.md]: I switched to `AutoTokenizer` and tested that it works with the class change.

**Pull request merged**